API/Entities/RateLimitOverrides.cs

```diff
@@ -0,0 +1,13 @@
+namespace API.Entities;
+
+public class RateLimitOverrides
+{
+    public RateLimitOverrides() { }
+    public RateLimitOverrides(int? permitLimit, int? window)
+    {
+        PermitLimit = permitLimit;
+        Window = window;
+    }
+    public int? PermitLimit { get; set; }
+    public int? Window { get; set; }
```

Comment:
What is the reason for using two constructors? Column names need to be defined on the properties, the migration uses pascal case for this table.

Reply:
So I was modeling my use of JSON columns off of the Microsoft [documentation](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#json-columns) and npgSQL [documentation](https://www.npgsql.org/efcore/mapping/json.html?tabs=data-annotations%2Cpoco). In my head some of the wording alluded to needing a parameterized constructor in order for EF to handle automatic (de)serialization, but I could be wrong. I included a parameter-less constructor for ease of partial updates. I was not sure if former was required to exist, but I should have tested instead of leaving it in.

Reply:
Sure, lmk what you find. Preferably we don't ever want to use constructors on entities. And test the column naming as well (you can verify with `dotnet ef migrations script` before applying to the db).

Reply:
I was going to raise the point that technically this doesn't conform to our definition of what an entity is and I was unsure where to put it. Maybe `/Entities/JsonElements` ? Since it is a nullable object field, the thought is that it would be easily assignable to a user via something like
```cs
if (user.RateLimitOverrides is null)
{
    user.RateLimitOverrides = new() { TokenLimit = 60 }
}
```
However the other side of that coin is that we could just make the `User.RateLimitOverrides` field non-null and just default to having null fields instead, but I thought that saving space in the db would be more important